modules/reactor/src/meshgenerators/PatternedHexMeshGenerator.C

```diff
@@ -0,0 +1,969 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "PatternedHexMeshGenerator.h"
+
+// C++ includes
+#include <cmath> // provides round, not std::round (see http://www.cplusplus.com/reference/cmath/round/)
+#include <fstream> // used to generate the optional control drum position file
+
+registerMooseObject("ReactorApp", PatternedHexMeshGenerator);
+
+defineLegacyParams(PatternedHexMeshGenerator);
+
+InputParameters
+PatternedHexMeshGenerator::validParams()
+{
+  InputParameters params = PolygonMeshGeneratorBase::validParams();
```

Comment:
there s a lot of parameters here. Could you please use the grouping parameter feature?
See interfaces/OutputInterface.C for an example

Reply:
same for other classes btw